src/documents/index.py

```diff
@@ -171,91 +175,88 @@ def remove_document_from_index(document):
                 )
-            elif k == "is_tagged":
-                criterias.append(query.Term("has_tag", self.evalBoolean(v)))
-            elif k == "document_type__isnull":
-                criterias.append(query.Term("has_type", self.evalBoolean(v) is False))
-            elif k == "created__date__lt":
+            elif query_filter == "id__all":
+                for xid in v.split(","):
+                    criterias.append(query.Term(f"{field}_id", xid))
+
+            elif query_filter == "date__lt":
+                criterias.append(
+                    query.DateRange(field, start=None, end=isoparse(v)),
+                )
+            elif query_filter == "date__gt":
+                criterias.append(
+                    query.DateRange(field, start=isoparse(v), end=None),
+                )
+            elif query_filter == "icontains":
                 criterias.append(
-                    query.DateRange("created", start=None, end=isoparse(v)),
+                    query.Term(field, v),
                 )
-            elif k == "created__date__gt":
+            elif query_filter == "istartswith":
                 criterias.append(
-                    query.DateRange("created", start=isoparse(v), end=None),
+                    query.Prefix(field, v),
                 )
-            elif k == "added__date__gt":
-                criterias.append(query.DateRange("added", start=isoparse(v), end=None))
-            elif k == "added__date__lt":
-                criterias.append(query.DateRange("added", start=None, end=isoparse(v)))
-            elif k == "storage_path__id":
-                criterias.append(query.Term("path_id", v))
-            elif k == "storage_path__id__in":
-                storage_paths_in = []
-                for storage_path_id in v.split(","):
-                    storage_paths_in.append(query.Term("path_id", storage_path_id))
-                criterias.append(query.Or(storage_paths_in))
-            elif k == "storage_path__id__none":
-                for storage_path_id in v.split(","):
-                    criterias.append(query.Not(query.Term("path_id", storage_path_id)))
-            elif k == "storage_path__isnull":
-                criterias.append(query.Term("has_path", self.evalBoolean(v) is False))
-            elif k == "owner__isnull":
-                criterias.append(query.Term("has_owner", self.evalBoolean(v) is False))
-            elif k == "owner__id":
-                criterias.append(query.Term("owner_id", v))
-            elif k == "owner__id__in":
-                owners_in = []
-                for owner_id in v.split(","):
-                    owners_in.append(query.Term("owner_id", owner_id))
-                criterias.append(query.Or(owners_in))
-            elif k == "owner__id__none":
-                for owner_id in v.split(","):
-                    criterias.append(query.Not(query.Term("owner_id", owner_id)))
```

Comment:
Thanks for updating this but yea, I do think the refactor is a bit overkill. I dont find this particularly more readable (rather, I find it less much less readable especially with things like "KeyError Unable to build a query filter") since this is a totally made up structure and it is still full of various conditionals.

If you wouldn't mind reverting this to the previous logic and just adding the changes necessary for the new filters that would be great. Of course I can do that if you'd prefer / if you dont have the time.